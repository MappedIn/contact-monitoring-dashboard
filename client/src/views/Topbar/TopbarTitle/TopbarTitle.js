import React, { PureComponent } from 'react';

import './TopbarTitle.scss';

const url = window.location.hostname || window.location.host || '';

let mode;
if (url.indexOf('qa') !== -1) {
	mode = 'QA';
} else if (url.indexOf('staging') !== -1) {
	mode = 'Staging';
} else if (url.indexOf('localhost') !== -1) {
	mode = 'Development';
}

class TopbarTitle extends PureComponent {
	static get defaultProps() {
		return {
			mode,
		};
	}

	render() {
		const { title, titleColor, mode } = this.props;

		const color = mode ? '#ffffff' : titleColor;

		return (
			<a
				href="/"
				className="topbar-title"
				style={{
					color: color,
					fill: color,
				}}>
				<span className="topbar-title-logo-text-container">
					<span className="topbar-title-logo-container">
						<svg
							className="topbar-title-logo"
							xmlns="http://www.w3.org/2000/svg"
							viewBox="0 0 116.8 185">
							<path d="M109.1,56.6H89.3c-5,0-7.4,2.2-7.4,7v60.3l-40.3-62c-2.8-4.3-4.6-5.1-10.2-5.3H11.9c-5.2,0-7.6,2.2-7.6,7V178.1c0,4.8,2.4,7,7.6,7H31.7c5.2,0,7.6-2.2,7.6-7v-60l40.5,61.7c3,4.5,4.6,5.3,10.2,5.3h19.2c5.2,0,7.6-2.2,7.6-7V63.6C116.6,58.8,114.2,56.6,109.1,56.6Z" />
							<circle cx="21.8" cy="21.8" r="21.8" />
						</svg>
					</span>

					<span className="topbar-title-text">{title}</span>
				</span>
				{mode && <span className="topbar-title-mode">{mode}</span>}
			</a>
		);
	}
}

export default TopbarTitle;
