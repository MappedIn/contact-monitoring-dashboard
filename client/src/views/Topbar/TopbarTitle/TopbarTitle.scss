.topbar-title {
	flex: 0 0 auto;

	display: flex;
	flex-flow: column nowrap;

	color: #ff5000;
	fill: #ff5000;
	margin-right: 20px;
	cursor: pointer;

	.topbar-title-logo-text-container {
		flex: 0 0 20px;
		height: 20px;

		display: flex;
		flex-flow: row nowrap;
		align-items: center;
		justify-content: center;
	}

	.topbar-title-logo-container {
		height: 20px;
		width: 20px;
		margin-right: 6px;
		overflow: visible;
		display: flex;
		align-items: flex-end;
	}

	.topbar-title-logo {
		width: 18px;
		height: 29px;
	}

	.topbar-title-text {
		font-size: 25px;
		font-weight: 600;

		overflow: hidden;
		white-space: nowrap;

		vertical-align: text-bottom;
		align-self: center;
	}

	.topbar-title-mode {
		flex: 0 0 12px;
		height: 12px;
		font-size: 12px;
		margin-top: 1px;
		margin-bottom: -6px;
	}
}
