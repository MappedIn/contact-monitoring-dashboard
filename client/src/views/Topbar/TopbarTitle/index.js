import TopbarTitle from './TopbarTitle';
export default TopbarTitle;
