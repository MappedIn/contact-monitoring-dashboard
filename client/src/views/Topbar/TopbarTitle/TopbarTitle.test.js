import React from 'react';
import renderer from 'react-test-renderer';

import TopbarTitle from './TopbarTitle';

test('TopbarTitle renders defaults', () => {
	const component = renderer.create(<TopbarTitle />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('TopbarTitle renders title, titleColor, and mode', () => {
	const component = renderer.create(
		<TopbarTitle title="Title" titleColor="red" mode="test" />,
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
