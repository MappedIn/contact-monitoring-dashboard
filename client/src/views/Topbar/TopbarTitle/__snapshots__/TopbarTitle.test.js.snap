// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarTitle renders defaults 1`] = `
<a
  className="topbar-title"
  href="/"
  style={
    Object {
      "color": "#ffffff",
      "fill": "#ffffff",
    }
  }
>
  <span
    className="topbar-title-logo-text-container"
  >
    <span
      className="topbar-title-logo-container"
    >
      <svg
        className="topbar-title-logo"
        viewBox="0 0 116.8 185"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M109.1,56.6H89.3c-5,0-7.4,2.2-7.4,7v60.3l-40.3-62c-2.8-4.3-4.6-5.1-10.2-5.3H11.9c-5.2,0-7.6,2.2-7.6,7V178.1c0,4.8,2.4,7,7.6,7H31.7c5.2,0,7.6-2.2,7.6-7v-60l40.5,61.7c3,4.5,4.6,5.3,10.2,5.3h19.2c5.2,0,7.6-2.2,7.6-7V63.6C116.6,58.8,114.2,56.6,109.1,56.6Z"
        />
        <circle
          cx="21.8"
          cy="21.8"
          r="21.8"
        />
      </svg>
    </span>
    <span
      className="topbar-title-text"
    />
  </span>
  <span
    className="topbar-title-mode"
  >
    Development
  </span>
</a>
`;

exports[`TopbarTitle renders title, titleColor, and mode 1`] = `
<a
  className="topbar-title"
  href="/"
  style={
    Object {
      "color": "#ffffff",
      "fill": "#ffffff",
    }
  }
>
  <span
    className="topbar-title-logo-text-container"
  >
    <span
      className="topbar-title-logo-container"
    >
      <svg
        className="topbar-title-logo"
        viewBox="0 0 116.8 185"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M109.1,56.6H89.3c-5,0-7.4,2.2-7.4,7v60.3l-40.3-62c-2.8-4.3-4.6-5.1-10.2-5.3H11.9c-5.2,0-7.6,2.2-7.6,7V178.1c0,4.8,2.4,7,7.6,7H31.7c5.2,0,7.6-2.2,7.6-7v-60l40.5,61.7c3,4.5,4.6,5.3,10.2,5.3h19.2c5.2,0,7.6-2.2,7.6-7V63.6C116.6,58.8,114.2,56.6,109.1,56.6Z"
        />
        <circle
          cx="21.8"
          cy="21.8"
          r="21.8"
        />
      </svg>
    </span>
    <span
      className="topbar-title-text"
    >
      Title
    </span>
  </span>
  <span
    className="topbar-title-mode"
  >
    test
  </span>
</a>
`;
